backfill/2022-01-27-sanitize-search-engines/org_mozilla_fenix.sql

```diff
@@ -0,0 +1,149 @@
+CREATE TEMP FUNCTION sanitize_labeled_counter(input ARRAY<STRUCT<key STRING, value INT64>>) AS (
+  ARRAY(
+    WITH base AS (
+      SELECT
+        *,
+        REGEXP_EXTRACT(key, r"([^.]+[.]in-content[.][^.]+[.])[^.]*[.]?.*") AS prefix,
+        REGEXP_EXTRACT(key, r"[^.]+[.]in-content[.][^.]+[.]([^.]*)[.]?.*") AS code,
+        REGEXP_EXTRACT(key, r"[^.]+[.]in-content[.][^.]+[.][^.]*[.]?(.*)") AS channel,
+      FROM
+        UNNEST(input)
+    ),
+    validations AS (
+      SELECT
+        *,
+        code IN (
+          "none",
+          "other",
+          "hz",
+          "h_",
+          "moz2",
+          "moz4",
+          "moz5",
+          "moza",
+          "mozb",
+          "mozd",
+          "moze",
+          "mozi",
+          "mozm",
+          "mozo",
+          "mozt",
+          "mozw",
+          "mozsl01",
+          "mozsl02",
+          "mozsl03",
+          "firefox-a",
+          "firefox-b",
+          "firefox-b-1",
+          "firefox-b-ab",
+          "firefox-b-1-ab",
+          "firefox-b-d",
+          "firefox-b-1-d",
+          "firefox-b-e",
+          "firefox-b-1-e",
+          "firefox-b-m",
+          "firefox-b-1-m",
+          "firefox-b-o",
+          "firefox-b-1-o",
+          "firefox-b-lm",
+          "firefox-b-1-lm",
+          "firefox-b-lg",
+          "firefox-b-huawei-h1611",
+          "firefox-b-is-oem1",
+          "firefox-b-oem1",
+          "firefox-b-oem2",
+          "firefox-b-tinno",
+          "firefox-b-pn-wt",
+          "firefox-b-pn-wt-us",
+          "ubuntu",
+          "ffab",
+          "ffcm",
+          "ffhp",
+          "ffip",
+          "ffit",
+          "ffnt",
+          "ffocus",
+          "ffos",
+          "ffsb",
+          "fpas",
+          "fpsa",
+          "ftas",
+          "ftsa",
+          "newext",
+          "monline_dg",
+          "monline_3_dg",
+          "monline_4_dg",
+          "monline_7_dg",
+          "_1000969a",
+          "_1000969b"
+        ) AS code_is_valid,
+        channel IS NULL
+        OR channel = ""
+        OR channel = "ts" AS channel_is_valid,
+      FROM
+        base
+    ),
+    scrubbed AS (
+      SELECT
+        * REPLACE (
+          CASE
+          -- If prefix is null, this didn't match the regex and we want to pass through unchanged
+          WHEN
+            prefix IS NULL
+          THEN
+            key
+          WHEN
+            code_is_valid
+            AND channel_is_valid
+          THEN
+            key
+          WHEN
+            code_is_valid
+            AND channel = "ts"
+          THEN
+            FORMAT("%s%s.%s", prefix, code, channel)
```

Comment:
This `WHEN` clause is duplicative of the previous `WHEN` clause, because if `channel = "ts"` then `channel_is_valid = TRUE`, so this code path will never be taken.

The [corresponding code path in `MessageScrubber.java`](https://github.com/mozilla/gcp-ingestion/blob/b88da84c017ee6251947b5223019bee90d20ea3b/ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/MessageScrubber.java#L475-L480) is actually never taken either, for the same reason.

What's missing is handling when the code is valid but the channel isn't:
```suggestion
          WHEN
            code_is_valid
            AND NOT channel_is_valid
          THEN
            FORMAT("%s%s", prefix, code)
```